File: ExampleProblems/Week 2/Problem/JavaScript/Queue.js
// Example program that demonstrates the use of a Queue (using Array)
const queueExample = () => {
  // Create a Queue of strings
  let queue = [];

  // Add elements to the queue
  queue.push("Apple");
  queue.push("Banana");
  queue.push("Cherry");
  queue.push("Date");

  // Print the queue elements
  console.log("Queue elements:");
  console.log(queue);

  // Remove an element from the front of the queue
  let removedElement = queue.shift();
  console.log("Removed element:");
  console.log(removedElement);

  // Print the updated queue elements
  console.log("Updated queue elements:");
  console.log(queue);
};

queueExample();

File: ExampleProblems/Week 2/Problem/Python/Queue.py
# Example program that demonstrates the use of a Queue using deque from collections module
from collections import deque

def queue_example():
    # Create a Queue of strings
    queue = deque(["Apple", "Banana", "Cherry", "Date"])

    # Print the queue elements
    print("Queue elements:")
    print(queue)

    # Add an element to the queue
    queue.append("Elderberry")
    print("Updated queue elements:")
    print(queue)

    # Remove an element from the front of the queue
    removed_element = queue.popleft()
    print("Removed element:")
    print(removed_element)

    # Print the updated queue elements
    print("Updated queue elements:")
    print(queue)

queue_example()

